src/components/ScholarshipAmountField.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import {
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  TextField,
+} from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import PropTypes from 'prop-types';
+import AmountType from '../types/AmountType';
+
+const useStyles = makeStyles({
+  formControl: {
+    minWidth: 200,
+  },
+});
+
+function ScholarshipAmountField(props) {
+  const classes = useStyles();
+  const {
+    amountType,
+    minAmount,
+    maxAmount,
+    onTypeChange,
+    onMinChange,
+    onMaxChange,
+    updateMinMax,
+  } = props;
+
+  function displayAmountFields() {
+    if (amountType === AmountType.Fixed) {
+      return (
+        <TextField
+          label="Amount"
+          type="number"
+          InputLabelProps={{ shrink: true }}
+          fullWidth
+          InputProps={{ inputProps: { min: 0, step: 50 } }}
+          value={minAmount}
+          onChange={(e) => updateMinMax(parseInt(e.target.value, 10) || '')}
+        />
+      );
+    }
+    if (amountType === AmountType.Range) {
+      return ['minAmount', 'maxAmount'].map((field) => (
+        <TextField
+          key={field}
+          label={field}
+          type="number"
+          InputLabelProps={{ shrink: true }}
+          fullWidth
+          InputProps={{ inputProps: { min: 0, step: 50 } }}
+          value={field === 'minAmount' ? minAmount : maxAmount}
+          onChange={field === 'minAmount' ? onMinChange : onMaxChange}
+        />
+      ));
+    }
+    return <div />;
+  }
+
+  return (
+    <FormControl className={classes.formControl}>
+      <InputLabel shrink>Amount Type</InputLabel>
+      <Select value={amountType} onChange={onTypeChange}>
+        {Object.keys(AmountType).map((option) => {
```

Comment:
The EnumTextField component from the other PR would be useful here

Reply:
Alternatively instead of a drop-down you could also use four radio buttons with any related form elements next to them. The form elements can be disabled if their radio option isn't set.